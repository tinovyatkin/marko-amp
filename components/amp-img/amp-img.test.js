'use strict';

/* eslint-env browser */
const AmpImg = require('marko').load(require.resolve('./index.marko'));
const nestedAmpImg = require('marko').load(
  require.resolve('../../__tests__/templates/amp-img-nested.marko'),
);

describe('amp-img', () => {
  test('render <amp-img> when out.global.amp is true', () => {
    expect(
      AmpImg.renderToString({
        $global: {
          amp: true,
        },
        alt: 'Test alt',
        height: 200,
        src: 'https://localhost',
        width: 170,
      }),
    ).toMatchSnapshot();
  });

  test('render <img> when out.global.amp is not true', () => {
    expect(
      AmpImg.renderToString({
        alt: 'Test alt',
        height: 200,
        src: 'https://localhost',
        width: 170,
      }),
    ).toMatchSnapshot();
  });

  test('Nested template HTML render', () => {
    // render normal version
    expect(nestedAmpImg.renderToString()).toMatchSnapshot();
  });

  test('Nested template AMP render', () => {
    // render AMP version
    expect(
      nestedAmpImg.renderToString({
        $global: {
          amp: true,
        },
      }),
    ).toMatchSnapshot();
  });
});
