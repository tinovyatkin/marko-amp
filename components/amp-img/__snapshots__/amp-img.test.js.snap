// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`amp-img Nested template AMP render 1`] = `"<amp-img src=\\"images/mountains.webp\\" alt=\\"Mountains\\" height=\\"368\\" width=\\"550\\"><amp-img src=\\"images/mountains.jpg\\" alt=\\"Mountains\\" height=\\"368\\" width=\\"550\\" fallback></amp-img></amp-img>"`;

exports[`amp-img Nested template HTML render 1`] = `"<picture><source type=\\"image/webp\\" srcset=\\"images/mountains.webp\\"><img src=\\"images/mountains.jpg\\" alt=\\"Mountains\\" width=\\"550\\"></picture>"`;

exports[`amp-img render <amp-img> when out.global.amp is true 1`] = `"<amp-img src=\\"https://localhost\\" alt=\\"Test alt\\" height=\\"200\\" width=\\"170\\"></amp-img>"`;

exports[`amp-img render <img> when out.global.amp is not true 1`] = `"<img src=\\"https://localhost\\" alt=\\"Test alt\\" height=200 width=170>"`;
